### DISCUSSION TITLE
Using React bindings with user-generated HTML

### DISCUSSION THREAD

**Discussion by victordiaz** [floating-ui/floating-ui#2162] [Q&A] [answered]
I have the following problem:

There is some user-generated content in the form of HTML, which I sanitize in the backend.
When I bring the content to the UI, some of the `<a>` tags will need to trigger a tooltip on hover. 

My problem is that I don't know how to set the `refs.setReference` to all of those HTML `<a>` tags, so they can trigger the `refs.setFloating` ref.

Any hint would be greatly appreciated.

Thanks in advance and thanks for the wonderful library!

**Comment by atomiks** [comment, marked as answer]
I imagine you'd use `container.querySelector` to select and store the nodes, which can be passed to `refs.setReference()`.

- [Example A (single)](https://codesandbox.io/s/new-bash-vh3rwz?file=/src/App.tsx:0-567)
- [Example B (multiple)](https://codesandbox.io/s/admiring-leavitt-gmsw2b?file=/src/App.tsx)

**Reply by victordiaz** [reply]
Oh! thanks so much!
`refs.setReference()` is exactly what I needed.

Thanks for the examples, very simple and straight to the point.
All the best!
